numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -629,56 +649,49 @@ OBJECT_setitem(PyObject *op, void *ov, void *NPY_UNUSED(ap))
     return PyErr_Occurred() ? -1 : 0;
 }
 
+
 /* VOID */
 
 static PyObject *
 VOID_getitem(void *input, void *vap)
```

Comment:
How hard would it be to change `vap` to be a `dtype` instead of an `ndarray`? Would that remove the need for these hacks?

Reply:
That is my vision for the future. For many we may also need to pass in `aligned`. But at least in principle this is visible/public API through `descriptor->f->getitem`.
While visible API, the SciPy uses `copyswap`, I am not sure anyone else uses anything.

Reply:
Does scipy call or implement copyswap? If we loosened the spec to "pass either an array or dtype", then the callers wouldn't notice - only the implementers, which I suspect is just us and `quaternion`.

Reply:
It calls `descr->f->copyswap(..., NULL)`...

Reply:
If that's all it does, then scipy won't even notice if we change that from a null array to a null dtype.

Reply:
True, the question is whether we want to be so bold and mess with this "public" API. The thing about it is mostly:
`descr->f->method` is a lot on the`descr`, I currently enivision `Py_TYPE(descr)->method` (or maybe one more indirection, similar to pythons `nb_...` slots). At which point we can deprecate/hide `->f` completely, and finally remove it.

The downside being that even after replacing, we have to keep supporting them with these types of hacks for a (fairly long) time.

Reply:
What is the resolution: are we leaving this for later?

Reply:
I do not think I want to go there right now.